homeassistant/components/switch/flux.py

```diff
@@ -0,0 +1,227 @@
+        vol.All(vol.Coerce(int), vol.Range(min=1000, max=40000)),
+    vol.Optional(CONF_SUNSET_CT, default=3000):
+        vol.All(vol.Coerce(int), vol.Range(min=1000, max=40000)),
+    vol.Optional(CONF_STOP_CT, default=1900):
+        vol.All(vol.Coerce(int), vol.Range(min=1000, max=40000)),
+    vol.Optional(CONF_BRIGHTNESS, default=90):
+        vol.All(vol.Coerce(int), vol.Range(min=0, max=255))
+})
+
+
+def set_lights_xy(hass, lights, x_val, y_val, brightness):
+    """Set color of array of lights."""
+    for light in lights:
+        if is_on(hass, light):
+            turn_on(hass, light,
+                    xy_color=[x_val, y_val],
+                    brightness=brightness)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the demo switches."""
+    name = config[CONF_NAME]
+    lights = config[CONF_LIGHTS]
+    start_time = config[CONF_START_TIME]
+    stop_time = config[COCONF_STOP_TIME]
+    start_colortemp = config[CONF_START_CT]
+    sunset_colortemp = config[CONF_SUNSET_CT]
+    stop_colortemp = config[CONF_STOP_CT]
+    brightness = config[CONF_BRIGHTNESS]
+    flux = FluxSwitch(name, hass, False, None, lights, start_time, stop_time,
+                      start_colortemp, sunset_colortemp, stop_colortemp,
+                      brightness)
+    add_devices([flux])
+
+    def update(call=None):
+        """Update lights."""
+        flux.flux_update()
+
+    hass.services.register(DOMAIN, 'flux_update', update)
+
+
+# pylint: disable=too-many-instance-attributes
+class FluxSwitch(SwitchDevice):
+    """Flux switch."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, name, hass, state, icon, lights, start_time, stop_time,
+                 start_colortemp, sunset_colortemp, stop_colortemp,
+                 brightness):
+        """Initialize the Flux switch."""
+        self._name = name
+        self.hass = hass
+        self._state = state
+        self._icon = icon
+        self._assumed = False
+        self._lights = lights
+        self._start_time = start_time
+        self._stop_time = stop_time
+        self._start_colortemp = start_colortemp
+        self._sunset_colortemp = sunset_colortemp
+        self._stop_colortemp = stop_colortemp
+        self._brightness = brightness
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo switch."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use for device if any."""
+        return self._icon
+
+    @property
+    def assumed_state(self):
+        """Return if the state is based on assumptions."""
+        return self._assumed
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._state
+
+    @property
```

Comment:
I don't see any real use for these properties, why not just refer the attributes directly. Properties should be used to expose things externally but no one is using this class.